R/tokens_replace.R

```diff
@@ -73,10 +74,14 @@ tokens_replace.tokens_xptr <- function(x, pattern, replacement, valuetype = "glo
 
     if (is.null(apply_if))
         apply_if <- rep(TRUE, length.out = ndoc(x))
+    if (verbose)
+        before <- stats_tokens(x)
     result <- cpp_tokens_replace(x, ids_pat, ids_rep[attr(ids_pat, "pattern")], !apply_if,
```

Comment:
@koheiw This is the line throwing the error. It seems to happen when:
1) `verbose = TRUE` (as in the test that throws the error)
2)`verbose = FALSE` but I put a breakpoint and run the function before it executes in the code.

Something to do with the mysteries of xptrs. Can you fix this?

Reply:
It was causing error because `set_types() <-`  was placed before `stats_tokens()`, where `ntoken()` is called. `ntoken()` calls the recompilation function, so unused tokens `TAX` is removed from the types in the XPtr object. Despite this, `cpp_tokens_replace()` replaced matched tokens with token ID for `TAX` and returned invalid tokens object.